sdk/cognitiveservices/azure-cognitiveservices-formrecognizer/DESIGN_DOC.md

```diff
@@ -0,0 +1,436 @@
+    status: str
+    pages: int
+    document_errors: List[ErrorInformation]
+
+class FieldNames:
+    field_name: str
+    accuracy: float
+
+class ErrorInformation:
+    code: str
+    message: str
+
+# Analyze ---------------------------------------------------
+class LabeledExtractedPage:
+    fields: List[ExtractedLabel]
+    tables: List[ExtractedTable]
+    page_number: int
+
+class ExtractedLabel:
+    name: str
+    value: LabelValue
+
+class LabelValue:
+    text: str
+    bounding_box: List[float]
+    confidence: float
+    raw_field: List[ExtractedLine]
+
+class ExtractedLine:
+    text: str
+    bounding_box: List[float]
+    language: str
+    words: List[ExtractedWord]
+
+class ExtractedWord:
+    text: str
+    bounding_box: List[float]
+    confidence: float
+```
+
+### List/Get/Delete Models
+```python
+class ModelInfo:
+    model_id: str
+    status: str
+    created_date_time: ~datetime.datetime
+    last_updated_date_time: ~datetime.datetime
+
+class ModelsSummary:
+    count: int
+    limit: int
+    last_updated_date_time: ~datetime.datetime
+```
+
+
+### Custom Training Samples
+
+#### Custom: Train and Analyze without labels
+```python
+from azure.ai.formrecognizer import CustomFormClient
+
+client = CustomFormClient(endpoint=endpoint, credential=credential)
+
+blob_sas_url = "xxxxx"  # training documents uploaded to blob storage
+poller = client.begin_training(blob_sas_url)
+custom_model = poller.result()
+
+print("Model ID: {}".format(custom_model.model_id))
+print("Extracted fields/clustered: {}".format(custom_model.form_clusters))
+
+# Check the training results
+for document in custom_model.train_result.documents:
+    print(document.name)
+    print(document.status)
+    print(document.pages)
+    print(document.document_errors)
+
+blob_sas_url = "xxxxx"  # form to analyze uploaded to blob storage
+poller = client.begin_extract_form(blob_sas_url, model_id=custom_model.model_id)
+result = poller.result()
+
+for page in result:
+    print("Page: {}".format(page.page_number))
+    for field in page.fields:
+        print(field.name.text, field.value.text)
+```
+
+#### Train and Analyze with labels
+```python
+from azure.ai.formrecognizer import CustomFormClient
+
+client = CustomFormClient(endpoint=endpoint, credential=credential)
+
+blob_sas_url = "xxxxx"  # training documents uploaded to blob storage
+poller = client.begin_labeled_training(blob_sas_url)
+custom_model = poller.result()
+
+print("Model ID: {}".format(custom_model.model_id))
+print("List of fields/accuracy: {}".format(custom_model.train_result.fields))
+
+# Check the training results
+for document in custom_model.train_result.documents:
+    print(document.name)
+    print(document.status)
+    print(document.pages)
+    print(document.document_errors)
+
+blob_sas_url = "xxxxx"  # form to analyze uploaded to blob storage
+poller = client.begin_extract_labeled_fields(blob_sas_url, model_id=custom_model.model_id)
+result = poller.result()
```

Comment:
How do I get the train result later? Should we have a `client.get_training_result(model_id)` method?

Reply:
Can you speak to the user scenario a little bit here?

Our guidelines say that the poller object should be restart-able to get the result. Is there a scenario of needing to get the train result later (when poller obj is not available)? If so, I'll definitely add this method.

Reply:
There is a lot of model metadata on this result object. To name just one scenario, I might want to generate a report of model accuracies for all of my models.

From reading the C# guidelines, it sounds like I can call `client.begin_extract_labeled_fields(model_id).get_value()` to retrieve the model metadata. This might work, but it's pretty cumbersome for a simple GET operation, and it's not intuitive that `begin_xyz` should be called to get an existing resource.

Reply:
Thanks for the insight. I will add this method to design doc and we can discuss more at arch board.